src/input_xml.F90

```diff
@@ -5280,79 +5271,51 @@ subroutine assign_temperatures(material_temps)
-      if (nuc % name == nuclides_0K(i) % nuclide) then
-        ! Copy basic information from settings.xml
+    do i = 1, size(res_scat_nuclides)
+      if (nuc % name == res_scat_nuclides(i)) then
+        ! Set nuclide to be resonant
         nuc % resonant = .true.
-        nuc % scheme = trim(nuclides_0K(i) % scheme)
-        nuc % E_min = nuclides_0K(i) % E_min
-        nuc % E_max = nuclides_0K(i) % E_max
-
-        ! Get index in libraries array
-        name = nuc % name
-        i_library = library_dict % get_key(to_lower(name))
-
-        call write_message('Reading ' // trim(name) // ' 0K data from ' // &
-             trim(libraries(i_library) % path), 6)
-
-        ! Open file and make sure version matches
-        file_id = file_open(libraries(i_library) % path, 'r')
-
-        ! Read nuclide data from HDF5
-        group_id = open_group(file_id, name)
-        method = TEMPERATURE_NEAREST
-        call resonant_nuc % from_hdf5(group_id, temperature, &
-             method, 1000.0_8, master)
-        call close_group(group_id)
-        call file_close(file_id)
-
-        ! Copy 0K energy grid and elastic scattering cross section
-        call move_alloc(TO=nuc % energy_0K, FROM=resonant_nuc % grid(1) % energy)
-        call move_alloc(TO=nuc % elastic_0K, FROM=resonant_nuc % sum_xs(1) % elastic)
-        nuc % n_grid_0K = size(nuc % energy_0K)
 
         ! Build CDF for 0K elastic scattering
         xs_cdf_sum = ZERO
         allocate(nuc % xs_cdf(size(nuc % energy_0K)))
 
-        do j = 1, size(nuc % energy_0K) - 1
-          ! Negative cross sections result in a CDF that is not monotonically
-          ! increasing. Set all negative xs values to ZERO.
-          if (nuc % elastic_0K(j) < ZERO) nuc % elastic_0K(j) = ZERO
-
-          ! build xs cdf
-          xs_cdf_sum = xs_cdf_sum &
-               + (sqrt(nuc % energy_0K(j)) * nuc % elastic_0K(j) &
-               + sqrt(nuc % energy_0K(j+1)) * nuc % elastic_0K(j+1)) / TWO &
-               * (nuc % energy_0K(j+1) - nuc % energy_0K(j))
-          nuc % xs_cdf(j) = xs_cdf_sum
-        end do
+        associate (E => nuc % energy_0K, xs => nuc % elastic_0K)
+          do j = 1, size(E) - 1
+            ! Negative cross sections result in a CDF that is not monotonically
+            ! increasing. Set all negative xs values to zero.
+            if (xs(j) < ZERO) xs(j) = ZERO
+
+            ! build xs cdf
+            xs_cdf_sum = xs_cdf_sum + (sqrt(E(j))*xs(j) + sqrt(E(j+1))*xs(j+1))&
+                 / TWO * (E(j+1) - E(j))
+            nuc % xs_cdf(j) = xs_cdf_sum
+          end do
+        end associate
+
+        ! Check to make sure resonance scattering upper energy is below URR
+        ! minimum, if present
+        if (nuc % urr_present) then
+          if (res_scat_energy_max > nuc % urr_data(1) % energy(1)) then
+            call fatal_error("Resonance scattering maximum energy is above the &
+                 &bottom of the unresolved resonance region.")
```

Comment:
Good point. It looks like resonant scattering [won't be used in the URR](https://github.com/mit-crpg/openmc/blob/4353e57f956de3b4baba77e38d50ddd49668efc2/src/physics.F90#L465-L470) anyway.